cmds/cpu/cpu.go

```diff
@@ -149,17 +159,50 @@ func newCPU(host string, args ...string) error {
 	if err := c.Dial(); err != nil {
 		return fmt.Errorf("Dial: %v", err)
 	}
-	verbose("start")
-	if err := c.Start(); err != nil {
-		return fmt.Errorf("Start: %v", err)
-	}
-	verbose("wait")
-	if err := c.Wait(); err != nil {
-		log.Printf("Wait: %v\r\n", err)
+
+	sigChan := make(chan os.Signal, 1)
+	defer close(sigChan)
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+	defer signal.Stop(sigChan)
+	errChan := make(chan error, 1)
+	defer close(errChan)
+
+	go func() {
+		verbose("start")
+		if err := c.Start(); err != nil {
+			errChan <- fmt.Errorf("Start: %v", err)
+			return
+		}
+		verbose("wait")
+		if err := c.Wait(); err != nil {
+			errChan <- fmt.Errorf("Wait: %v\r\n", err)
+		} else {
+			errChan <- nil
+		}
+	}()
+
+	var err error
+loop:
+	for {
+		select {
+		case sig := <-sigChan:
+			var sigErr error
+			switch sig {
+			case syscall.SIGINT:
+				sigErr = c.Signal(ossh.SIGINT)
+			case syscall.SIGTERM:
+				sigErr = c.Signal(ossh.SIGTERM)
+			}
+			verbose("signal %v sent to %q", sig, c.Args[0])
+			if sigErr != nil {
+				err = fmt.Errorf("sending %v to %q: %v", sig, c.Args[0], sigErr)
```

Comment:
I would argue that we should not quit here, log an error and continue?

Reply:
Done.